cargo/workspace.bzl

```diff
@@ -0,0 +1,251 @@
+    ]
+
+    result = repository_ctx.execute(
+        args,
+        environment = {
+            "RUST_LOG": "info",
+
+            # The resolver invokes `cargo metadata` which relies on `rustc` being on the $PATH
+            "PATH": ":".join([str(b.dirname) for b in path_binaries]),
+        },
+    )
+    repository_ctx.delete(input_path)
+    if result.stderr:
+        print("Output from resolver: " + result.stderr)
+    if result.return_code != 0:
+        fail("Error resolving deps:\n" + result.stdout + "\n" + result.stderr)
+
+    repository_ctx.file("BUILD.bazel")
+
+_crate_universe_resolve = repository_rule(
+    implementation = _crate_universe_resolve_impl,
+    attrs = {
+        "packages": attr.string_list(allow_empty = True),
+        "cargo_toml_files": attr.label_list(),
+        "overrides": attr.string_dict(doc = """Mapping of crate name to crate.override(...) entries)
+
+Example:
+
+    load("@rules_rust//cargo:workspace.bzl", "rust_library", "crate")
+
+    rust_library(
+        name = "mylib",
+        # [...]
+        overrides = {
+            "tokio": crate.override(
+                extra_rust_env_vars = {
+                    "MY_ENV_VAR": "MY_ENV_VALUE",
+                },
+                extra_build_script_env_vars = {
+                    "MY_BUILD_SCRIPT_ENV_VAR": "MY_ENV_VALUE",
+                },
+                extra_bazel_deps = {
+                    # Extra dependencies are per target. They are additive.
+                    "cfg(unix)": ["@somerepo//:foo"],  # cfg() predicate.
+                    "x86_64-apple-darwin": ["@somerepo//:bar"],  # Specific triple.
+                    "cfg(all())": ["@somerepo//:baz"],  # Applies to all targets ("regular dependency").
+                },
+                extra_build_script_bazel_deps = {
+                    # Extra dependencies are per target. They are additive.
+                    "cfg(unix)": ["@buildscriptdep//:foo"],
+                    "x86_64-apple-darwin": ["@buildscriptdep//:bar"],
+                    "cfg(all())": ["@buildscriptdep//:baz"],
+                },
+                extra_bazel_data_deps = {
+                    # ...
+                },
+                extra_build_script_bazel_data_deps = {
+                    # ...
+                },
+            ),
+        },
+    )
+
+"""),
+        "repository_template": attr.string(),
+        "supported_targets": attr.string_list(allow_empty = False),
+        "lockfile": attr.label(
+            allow_single_file = True,
+            mandatory = False,
+        ),
+        "_resolver_script_linux": attr.label(
+            default = "@crate_universe_resolver_linux//file:downloaded",
+            executable = True,
+            cfg = "host",
+        ),
+        "_resolver_script_darwin": attr.label(
+            default = "@crate_universe_resolver_darwin//file:downloaded",
+            executable = True,
+            cfg = "host",
+        ),
+    },
+    local = True,
```

Comment:
Why is this a local rule? This will cause the rule to run every time, right? I would imagine this should not be local and if users need to rerun it, they can run `bazel sync --only=my_cargo_universe`

Reply:
> Why is this a local rule?
Good question! I don't think there's any reason for it to be a local rule, and so it no longer is!

> This will cause the rule to run every time, right?
No, it definitely doesn't re-run every time, I _think_ it will only re-run each time you explicitly `bazel fetch`, or if the `WORKSPACE` changes? But either way, removed!